client/src/modules/shopping-list/model/actions.js

```diff
@@ -173,7 +199,52 @@ export const updateList = (listId, data) => dispatch => {
       createNotificationWithTimeout(
         dispatch,
         NotificationType.ERROR,
-        err.message || "Oops, we're sorry, updating lists failed..."
+        err.message || "Oops, we're sorry, updating list failed..."
+      );
+    });
+};
+
+export const archiveList = (listId, isArchived) => dispatch => {
+  dispatch(archiveListRequest());
+  patchData(`${ENDPOINT_URL}/shopping-lists/${listId}/update`, { isArchived })
+    .then(resp => resp.json())
+    .then(json => {
+      dispatch(archiveListSuccess({ isArchived, listId }));
+      createNotificationWithTimeout(
+        dispatch,
+        NotificationType.SUCCESS,
+        'List was successfully archived!' || json.message
+      );
+    })
+    .catch(err => {
+      dispatch(archiveListFailure());
+      createNotificationWithTimeout(
+        dispatch,
+        NotificationType.ERROR,
+        err.message || "Oops, we're sorry, archiving list failed..."
+      );
+    });
+};
+
+export const restoreList = (listId, isArchived) => dispatch => {
+  dispatch(restoreListRequest());
+  patchData(`${ENDPOINT_URL}/shopping-lists/${listId}/update`, { isArchived })
+    .then(() => getData(`${ENDPOINT_URL}/shopping-lists/${listId}/data`))
```

Comment:
this seems a bit complicated for me. 
Can't you just return updated list when pathingData and than update reducer with returned data, or update reducer just using id of the list and toggle back isArchived property instead of replacing whole list object?

Reply:
We decided to not return data when documents are updating in db, we return status and message. I didn't want change this behaviour and modify other list's action. List data is removed from redux after list archivization so we have to get it one more time.